flang/include/flang/Tools/CLOptions.inc

```diff
@@ -117,10 +93,38 @@ inline void createMLIRToLLVMPassPipeline(mlir::PassManager &pm)
   pm.addNestedPass<mlir::FuncOp>(fir::createControlFlowLoweringPass());
   pm.addPass(mlir::createLowerToCFGPass());
 
+  mlir::GreedyRewriteConfig config;
+  config.enableRegionSimplification = false;
   pm.addPass(mlir::createCanonicalizerPass(config));
```

Comment:
Why use a config here but not above?

Reply:
Tl;Dr I don't know, I'm not the original author.

This code was extracted from `bbc` (from [here](https://github.com/flang-compiler/f18-llvm-project/blob/fir-dev/flang/tools/bbc/bbc.cpp#L270-L294)). This code in `bbc` was recently refactored in https://github.com/flang-compiler/f18-llvm-project/pull/1126.

@clementval ?